include/boost/gil/image_processing/morphology.hpp

```diff
@@ -0,0 +1,193 @@
+//
+// Copyright 2021 Prathamesh Tagore <prathameshtagore@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BOOST_GIL_IMAGE_PROCESSING_MORPHOLOGY_HPP
+#define BOOST_GIL_IMAGE_PROCESSING_MORPHOLOGY_HPP
+
+#include <boost/gil/extension/numeric/kernel.hpp>
+
+#include <boost/gil/gray.hpp>
+#include <boost/gil/algorithm.hpp>
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/image_processing/threshold.hpp>
+
+namespace gil = boost::gil;
+namespace boost{
+    namespace gil{
+        enum class morphological_operations
+            {
+                dilate,
+                erode,
+                opening,
+                closing,
+                morphological_gradient,
+                top_hat,
+                black_hat
+            };
+namespace detail{
+template <typename SrcView, typename DstView, typename Kernel>
+void morph_impl(SrcView const& src_view, DstView const& dst_view, Kernel const& kernel,const int identifier)
+{
+    int flip_ker_row, flip_ker_col, row_boundary, col_boundary;
+    typename channel_type<typename SrcView::value_type>::type max_overlapped_element,min_overlapped_element;
+    for (std::ptrdiff_t view_row = 0; view_row < src_view.height(); ++view_row)
+    {
+        for (std::ptrdiff_t view_col = 0; view_col < src_view.width(); ++view_col)
+        {
+            max_overlapped_element = (std::numeric_limits<typename channel_type<SrcView>::type>::min)();
+            min_overlapped_element = (std::numeric_limits<typename channel_type<SrcView>::type>::max)();
```

Comment:
Great catch, but unfortunately will not work for our quirky types like what you get when `SrcView == gil::gray32f_view_t` and the like. I would just set to the first element and mention in every function that uses this to not pass empty views, or bad things will happen.

Reply:
Okay,i will mention about not passing empty views in docs and in comments of example program(gil/example/morphology.cpp)